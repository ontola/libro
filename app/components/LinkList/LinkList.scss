@import '../shared/config';

.LinkList {
  -webkit-overflow-scrolling: touch; // Enables inertial scrolling on iOS
  display: flex;
  overflow-x: auto;
  white-space: nowrap; // Enables scrolling horizontally to reveal hidden items
}

.LinkList__link {
  border-bottom: 3px solid transparent;
  color: color('blue');
  display: block;
  margin-right: unit('large');
  padding: units('small', 'none');
  transition: .1s border-bottom;

  &:hover {
    border-bottom-color: color('blue', 'light');
  }

  &--active,
  &:active {
    border-bottom-color: color('blue');
    color: color('blue');
  }

  &:last-of-type {
    margin-right: unit('large');
  }
}
