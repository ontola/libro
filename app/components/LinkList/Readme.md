Basic LinkList

    const links = [{
      label: 'Link 1',
      to: '/',
    }, {
      label: 'Link 2',
      to: '/',
    }, {
      label: 'Link 3',
      to: '/',
    }, {
      label: 'Link 4',
      to: '/',
    }, {
      label: 'Link 5',
      to: '/',
    }, {
      label: 'Link 6',
      to: '/',
    }];
    <LinkList links={links} />

LinkList in context

    const links = [{
      label: 'Link with long title 1',
      to: '/',
    }, {
      label: 'Link with long title 2',
      to: '/',
    }, {
      label: 'Link with reasonably long title 3',
      to: '/',
    }, {
      label: 'Link with long title 4',
      to: '/',
    }, {
      label: 'Link with long title 5',
      to: '/',
    }, {
      label: 'Link with long title 6',
      to: '/',
    }];
    <Cover type="lighter">
      <Container>
        <LinkList links={links} fullWidth />
      </Container>
    </Cover>
