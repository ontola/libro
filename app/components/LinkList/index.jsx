import './LinkList.scss';
import React, { PropTypes } from 'react';
import { Link } from 'react-router';

const propTypes = {
  links: PropTypes.arrayOf(PropTypes.shape({
    to: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
  })).isRequired,
};

const defaultProps = {
  links: [],
};

const LinkList = ({ links }) => {
  const generateLinks = links.map((link, i) => (
    <Link
      onlyActiveOnIndex
      key={i}
      className="LinkList__link"
      activeClassName="LinkList__link--active"
      to={link.to}
      children={link.label}
    />
  ));

  return (
    <nav role="navigation" className="LinkList" children={generateLinks} />
  );
};

LinkList.propTypes = propTypes;
LinkList.defaultProps = defaultProps;

export default LinkList;
