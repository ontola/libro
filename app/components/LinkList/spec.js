import React from 'react';
import { mount } from 'enzyme';
import { assert } from 'chai';
import LinkList from './';

const links = [
  {
    label: 'Test 1',
    to: '/test1',
  },
  {
    label: 'Test 2',
    to: '/test2',
  },
];

describe('A navigation bar', () => {
  it('must contain one or more links', () => {
    const wrapper = mount(<LinkList links={links} />);
    const findLinks = wrapper.find('nav a');
    assert(findLinks.length > 0, 'No links have been found');
  });
});
