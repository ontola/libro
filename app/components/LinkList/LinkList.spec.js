import LinkList from './';

const exampleLinks = [{
  label: 'Test 1',
  to: '/test1',
}, {
  label: 'Test 2',
  to: '/test2',
}];

argUnit(LinkList, () => {
  set('links', () => exampleLinks);

  it('renders', () => {
    expect(subject).toMatchSnapshot();
  });

  it('renders all links', () => {
    expect(subject.find(marker('link'))).toHaveLength(exampleLinks.length);
  });
});
