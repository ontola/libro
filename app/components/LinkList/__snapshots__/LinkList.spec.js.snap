// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkList renders 1`] = `
<nav
  className="LinkList"
>
  <NavLink
    activeClassName="LinkList__link--active"
    aria-current="page"
    className="LinkList__link"
    data-test="LinkList-link"
    exact={true}
    key="Test 1-/test1"
    to="/test1"
  >
    Test 1
  </NavLink>
  <NavLink
    activeClassName="LinkList__link--active"
    aria-current="page"
    className="LinkList__link"
    data-test="LinkList-link"
    exact={true}
    key="Test 2-/test2"
    to="/test2"
  >
    Test 2
  </NavLink>
</nav>
`;
